geocatbridge/servers/models/geoserver.py

```diff
@@ -1101,13 +1104,40 @@ def clearWorkspace(self, recreate=True) -> bool:
                     else:
                         raise
 
+            # Get settings override
+            url = f"{self.apiUrl}/workspaces/{self.workspace}/settings.json"
+            workspace_settings = self.request(url).json()
+            # GeoServer returns *some* settings even if the workspace does not use any Workspace Specific Settings:
+            # https://osgeo-org.atlassian.net/browse/GEOS-11361
+            # Here we check if we got those exact settings and if so, consider them to mean "no settings".
+            # As of GeoServer 2.24.2:
+            default_workspace_settings = {
+                "settings": {
+                    "contact": {"id": "contact"},
+                    "charset": 'UTF-8',
+                    "numDecimals": 4,
+                    "verbose": False,
+                    "verboseExceptions": False,
+                    "localWorkspaceIncludesPrefix": False,
+                    "showCreatedTimeColumnsInAdminList": False,
+                    "showModifiedTimeColumnsInAdminList": False
+                }
+            }
+            if workspace_settings == default_workspace_settings:
+                self.logWarning(
+                    "Received *default* workspace specific settings which indicates "
+                    "that there were no *actual* settings. "
+                    "Setting the workspace to have *no* workspace specific settings."
+                )
+                workspace_settings = {}
+
```

Comment:
Hi @kannes, the question here is: does it hurt if you send back the "default" settings? It usually takes less than a second to do...

More importantly though: I think I have discovered how to distinct between "real" settings and no/default settings. If the `"settings"` contain an `"id"` property (usually set to something like `"SettingsInfoImpl--1b08c13b:18e5c7726dc:-7f54"` for example), you can safely assume that the workspace has actual settings turned on.

So what I would do is check for that `"id"` key and if you don't have it, do nothing.
If you do have it, copy the settings to the newly created workspace (but **remove that `"id"` key** before you `PUT` it - GeoServer will create a new `"id"` for it).

Reply:
I pushed code that uses the "id" now, thanks!